src/declarativetabmodel.cpp

```diff
@@ -249,7 +250,8 @@ const QList<Tab> &DeclarativeTabModel::tabs() const
 
 const Tab &DeclarativeTabModel::activeTab() const
 {
-    return m_activeTab;
+    Q_ASSERT(contains(m_activeTabId));
```

Comment:
Good point!
The assertion gets triggered when switching to the private mode at the function updating `DeclarativeWebContainer::m_tabId`. Now I wonder why we need to maintain `m_tabId` in `DeclarativeWebContainer`...